salt/modules/yumpkg.py

```diff
@@ -1056,9 +1056,10 @@ def install(name=None,
     return ret
 
 
-def upgrade(refresh=True, fromrepo=None, skip_verify=False, **kwargs):
+def upgrade(name=None, refresh=True, fromrepo=None, skip_verify=False, pkgs=None, normalize=True, **kwargs):
```

Comment:
The current api of fuction 'upgrade' is dangerous:

```
salt '*' pkg.upgrade openssl
```
is the same as
```
salt '*' pkg.upgrade refresh=openssl # doesn't make sense
```
and causes that all systems are *COMPLETELY* upgraded (all packages, not only openssl), because salt.utils.is_true('openssl') returns True. This happened to my colleague and caused serious troubles to the my whole team.

I can change the api of this function in all modules exporting virtual module pkg to address the API parity issue. What do you think about this? I think the principle of the least surprise isn't followed here. I understand the consequences of this change, but the worst-case scenario is that after upgrade to new SaltStack someone will run
```
salt '*' pkg.upgrade False # not so common I think
```
and will get 0 upgrades because package False doesn't exist. I think this is less evil than what happened to my team and can happen to anyone.